### PR TITLE
Distribution api

### PR DESCRIPTION
**Pull request by fritzo**
TODO Describe differences from pytorch master

### CODE REVIEW

File: torch/distributions.py [hunk 1]
```diff
@@ -31,41 +31,80 @@
...
+        return torch.exp(self.log_prob(value))
 
-        Args:
-            value (Tensor or Variable):
-        """
+    def log_cdf(self, value):
+        return torch.log(self.cdf(value))
+
+    def cdf(self, value):
+        raise NotImplementedError
```
Comment (fritzo) [inline torch/distributions.py]: Are these things you plan to implement before merging with upstream?

**Comment by fritzo** [comment]
FYI I've renamed `Multinomial` -> `Categorical` in https://github.com/pytorch/pytorch/pull/3766 to make way for an actual `Multinomial` distribution #4 . You may want to sync this branch.

**Comment by alicanb** [comment]
This was the branch that I was preparing to send upstream, before we weren't collaborating. The plan in my mind now, is to make this branch implement the changes in distribution class, mainly addressing  https://github.com/probtorch/pytorch/issues/7 and https://github.com/probtorch/pytorch/issues/8 , since these (especially https://github.com/probtorch/pytorch/issues/8) are blocking to rest.

**Comment by fritzo** [comment]
@alicanb I'd recommend breaking our work up into as small PRs as possible so as to minimize effort of code review for PyTorch folks. Specifically, I'd recommend submitting separate PRs for `.reparameterized`, `.shape`, etc. As I see it, the benefit of working on probtorch branches is that it can hide review latency for dependent tasks (e.g. if we implement a `random_gamma()` and then a `distributions.Gamma` on top of that).

**Comment by alicanb** [comment]
Sure, that can be arranged. I'll send a PR for `.reparameterized` in few minutes, we can build up from there.

**Pull request closed without merging**